src/main/java/core/basesyntax/service/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.exception.NotEnoughProductsException;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    private FruitStorageDao storageDao;
+
+    public PurchaseOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        int currentFruitQuantity = storageDao.getFruitQuantity(transaction.getFruit());
+        int purchaseResult = currentFruitQuantity - transaction.getQuantity();
+        if (purchaseResult < 0) {
+            throw new NotEnoughProductsException("Not enough fruits: "
+                        + transaction.getFruit()
+                        + ": "
+                        + transaction.getQuantity()
+                        + " available: "
+                        + currentFruitQuantity);
+        }
+        storageDao.getAllFruits().put(transaction.getFruit(), purchaseResult);
+    }
+}
```

Comment:
The class `PurchaseOperation` seems to be correctly implementing the `OperationHandler` interface. However, according to the checklist, you should check the result balance before saving it in the Storage to ensure it's positive. You've done this correctly in lines 16-24. Just make sure to rethrow a RuntimeException with an informative message and the exception object, which you have also done correctly.